### PR TITLE
Documentation only

### PR DESCRIPTION
**Pull request by BenjaminDev**
## Description
Addresses #107 
Added sphinx documentation [see example](https://benjamindev.github.io/PyDP/readme.html). This means we need to write docstrings in a sphinx compatible way. See `benjaminDev/pydp@distributions` pr for examples. This pr has a workflow that will run on pushes to  `dev` which will build and put the built docs in the `gh-pages` branch. 

## Affected Dependencies
Added dev dependencies: `sphinx` and `sphinx-rtd-theme`.

## How has this been tested?
- Tested on personal fork and it works their.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

## Todo:
This pr will has a workflow that will run on pushes to  `dev` which will put the built docs in the `gh-pages` branch. To serve these an Openmined developer must go to settings and activate github pages as shown below.

image

### CODE REVIEW

**Comment by chinmayshah99** [comment]
Hey @BenjaminDev ,
We already have readthedocs.org integration over here: https://pydp.readthedocs.io/en/latest/
So we just need to put everything in /docs folder. 

Or is this PR a better solution to that?

**Comment by BenjaminDev** [comment]
hahaha. Okay I missed that. I have never used read the docs. I didn't see a `.readthedocs.yml` file anywhere.

Can you point it to the gh-pages branch? or can readthedocs run the build command?d

I'd say the only issue is somewhere we need to build the docs as it'll get messy if we commit all the built files to github. We'd need to look into what readthedocs offers and whether sphinx can play nicely with sphinx.
Seems it can.
```
sphinx:
  builder: html
  configuration: conf.py
  fail_on_warning: true
```
Where is our .readthedocs.yml file? 

Is this current solution better? I'd say same same. But if other  OpenMined projects are using readthedocs service then let's do the same.

**Comment by chinmayshah99** [comment]
Can we add markdown support?

**Comment by BenjaminDev** [comment]
> Can we add markdown support?

I'm sure we can. which files are in markdown that need to be added? Unless they exist let's leave them to another PR. I'm sure there will be plenty to add.

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+python-dp
+sphinx-rtd-theme
```
Comment (chinmayshah99) [inline docs/requirements.txt:2]: Can we remove python-dp? or is that essential to the build?

Comment (chinmayshah99) [reply docs/requirements.txt:2]: Also, is this requirement file necessary?

**Comment by chinmayshah99** [comment]
Also since we have added readthedocs, can we remove the github action?

File: docs/requirements.txt (hunk 1 above)
Comment (BenjaminDev) [reply docs/requirements.txt:2]: The way sphinx works is by stripping the docs from the docstrings of the released (installed) package. Yes! this requirements.txt is needed and is used by reathedocs (or the github action proposed) when the docs are built.

**Comment by BenjaminDev** [comment]
> Also since we have added readthedocs, can we remove the github action?

Yeah we should.  I'll do that.

**Pull request merged**